src/coreclr/minipal/Unix/doublemapping.cpp

```diff
@@ -14,10 +14,10 @@
 #include <assert.h>
 #include <limits.h>
 #include <errno.h>
-#ifdef TARGET_LINUX
+#if defined(TARGET_LINUX) && !defined(ARM_SOFTFP)
```

Comment:
`ARM_SOFTFP` does not look like the most appropriate define to use for this context. Is there a better one to use?

cc @janvorli

Reply:
@clamp03 what is the build error that you were getting? I don't think we should base this check on the ARM_SOFTFP, since I don't believe it is related to memfd_create. I guess the issue is most likely a very recent distro that has something different w.r.t. the memfd_create.

Reply:
@janvorli 
Build error message (armel+tizen cross build) is
```
  [  2%] Building C object Native.Unix/System.IO.Compression.Native/CMakeFiles/System.IO.Compression.Native-Static.dir/home/dheon/work/dotnet/jobs/fnv/runtime/src/libraries/Native/AnyOS/brotli/common/platform.c.o
  /home/dheon/work/dotnet/jobs/fnv/runtime/src/coreclr/minipal/Unix/doublemapping.cpp:18:10: fatal error: 'linux/memfd.h' file not found 
  #include <linux/memfd.h>                                                                                                              
           ^~~~~~~~~~~~~~~                                                                                                                                                                                                                                                      
  1 error generated.
```
In our armel build environment (cross build rootfs), memfd_create and MFD_CLOEXEC are defined in headers included by ```sys/mman.h```. (others are not)

How about ```MFD_CLOEXEC``` instead of ```ARM_SOFTFP```? If ```MFD_CLOEXEC``` is defined in ```sys/mman.h``` already, it doesn't need to include ```linux/memfd.h```. Then I think it can support both old and very recent distro.
Thank you.

Reply:
That sounds like a good way.